libevm/VM.cpp

```diff
@@ -202,6 +202,24 @@ bytesConstRef VM::execImpl(u256& io_gas, ExtVMFace& _ext, OnOpFunc const& _onOp)
 		return nextPC;
 	};
 
+	auto copyDataToMemory = [](bytesConstRef _data, decltype(m_stack)& _stack, decltype(m_temp)& _memory)
+	{
+		auto offset = static_cast<size_t>(_stack.back());
+		_stack.pop_back();
+		bigint bigIndex = _stack.back();
+		auto index = static_cast<size_t>(bigIndex);
+		_stack.pop_back();
+		auto size = static_cast<size_t>(_stack.back());
+		_stack.pop_back();
+
+		auto sizeToBeCopied = bigIndex + size > _data.size() ? _data.size() < bigIndex ? 0 : _data.size() - index : size;
+
+		if (sizeToBeCopied > 0)
+			std::memcpy(_memory.data() + offset, _data.data() + index, sizeToBeCopied);
+		if (size - sizeToBeCopied > 0)
```

Comment:
Please don't use this in conjunction with `auto`. Depending on the signedness of the types of `size` and `sizeToBeCopied`, this might be equivalent to `size > sizeToBeCopied` or not.